go/vt/vtgate/planbuilder/operators/operator.go

```diff
@@ -35,6 +35,14 @@ type (
 	Operator interface {
 		Clone(inputs []Operator) Operator
 		Inputs() []Operator
+
+		// AddPredicate is used to push predicates. It pushed it as far down as is possible in the tree.
+		// If we encounter a join and the predicate depends on both sides of the join, the predicate will be split into two parts,
+		// where data is fetched from the LHS of the join to be used in the evaluation on the RHS
+		AddPredicate(ctx *plancontext.PlanningContext, expr sqlparser.Expr) (Operator, error)
+
+		// AddColumn tells an operator which columns it should output.
```

Comment:
`AddColumn tells an operator to also output an additional column specified. If the column already exists, its offset is returned.`
I think this comment makes more sense because AddColumn doesn't provide an exhaustive list. Like it doesn't say these are all the columns an operator needs to output.